nutanix/common_era_schema.go

```diff
@@ -417,3 +506,88 @@ func buildActionArgumentsFromResourceData(d *schema.Set, args []*era.Actionargum
 	}
 	return args
 }
+
+func buildSLADetails(pr []interface{}) *era.SLADetails {
+	if len(pr) > 0 {
+		res := &era.SLADetails{}
+
+		for _, v := range pr {
+			val := v.(map[string]interface{})
+
+			if priSLA, pok := val["primary_sla"]; pok {
+				res.PrimarySLA = expandPrimarySLA(priSLA.([]interface{}))
+			}
+		}
+		return res
+	}
+	return nil
+}
+
+func expandPrimarySLA(pr []interface{}) *era.PrimarySLA {
+	if len(pr) > 0 {
+		out := &era.PrimarySLA{}
+
+		for _, v := range pr {
+			val := v.(map[string]interface{})
+
+			if slaid, ok := val["sla_id"]; ok {
+				out.SLAID = utils.StringPtr(slaid.(string))
+			}
+
+			if nxcls, ok := val["nx_cluster_ids"]; ok {
+				res := make([]*string, 0)
+				nxclster := nxcls.([]interface{})
+
+				for _, v := range nxclster {
+					res = append(res, utils.StringPtr(v.(string)))
+				}
+				out.NxClusterIds = res
+			}
+		}
+		return out
+	}
+	return nil
+}
+
+func expandNodesProperties(pr *schema.Set) []*era.NodesProperties {
+	argSet := pr.List()
+
+	out := make([]*era.NodesProperties, 0)
+	for _, arg := range argSet {
+		var val interface{}
+		val = arg.(map[string]interface{})["value"]
+		b, ok := tryToConvertBool(arg.(map[string]interface{})["value"])
```

Comment:
what is purpose of this ?

Reply:
some values are string and others are boolean. So we check and append the argument .